core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -167,16 +170,23 @@ public void shutdown(Status status) {
       super.shutdown(status);
     }
 
-    // TODO(zivy@): should call delegate shutdownNow asap. Maybe cancel pending applier.
+    // TODO(zivy): cancel pending applier here.
     @Override
     public void shutdownNow(Status status) {
       checkNotNull(status, "status");
       synchronized (this) {
         if (pendingApplier.get() < 0) {
+          shutdownStatus = status;
+          if (savedShutdownStatus == null) {
```

Comment:
I don't think `savedShutdownStatus` needs to be set within shutdownNow(). I may be missing something. It doesn't hurt, but may be confusing.

Reply:
Oh I thought you meant "You may choose to have shutdownNow set the savedShutdownStatus if it is null."  Now I guess  "save the shutdownStatus" for the failing RPCs.
Let's remove it, too confusing.